ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp

```diff
@@ -75,6 +75,72 @@ void AirsimROSWrapper::initialize_statistics()
     // statistics_obj_ptr = {&setCarControlsStatistics, &getGpsDataStatistics, &getCarStateStatistics, &control_cmd_sub_statistics, &global_gps_pub_statistics, &odom_local_ned_pub_statistics};
 }
 
+void AirsimROSWrapper::publish_track() {
+    
+    CarApiBase::RefereeState state = airsim_client_.getRefereeState();
+    // Get car initial position
+    car_start_pos = state.car_start_location;
+
+    fsds_ros_bridge::Track track;
+    visualization_msgs::MarkerArray cone_markers;
+    uint32_t cone_id = 0;
+    for (const auto& cone : state.cones) {
+        fsds_ros_bridge::Cone cone_object;
+        cone_object.location.x = cone.location.x != 0 ? (cone.location.x - car_start_pos.x)*0.01 : 0;
+        cone_object.location.y = cone.location.y != 0 ? (cone.location.y - car_start_pos.y)*0.01 : 0;
+        if (cone.color == CarApiBase::ConeColor::Yellow) {
+            cone_object.color = fsds_ros_bridge::Cone::YELLOW;
+        } else if (cone.color == CarApiBase::ConeColor::Blue) {
+            cone_object.color = fsds_ros_bridge::Cone::BLUE;
+        } else if (cone.color == CarApiBase::ConeColor::OrangeLarge) {
+            cone_object.color = fsds_ros_bridge::Cone::ORANGE_BIG;
+        } else if (cone.color == CarApiBase::ConeColor::OrangeSmall) {
+            cone_object.color = fsds_ros_bridge::Cone::ORANGE_SMALL;
+        } else if (cone.color == CarApiBase::ConeColor::Unknown) {
+            cone_object.color = fsds_ros_bridge::Cone::UNKNOWN;
+        }
+        track.track.push_back(cone_object);
+        
+        // Fill Marker
+        visualization_msgs::Marker cone_marker;
+        // Note that because choosing this NED frame, Rviz will display a map which is flipped wrt the real map 
+        cone_marker.header.frame_id = "fsds/FSCar";
+        cone_marker.header.stamp = ros::Time::now();
+        cone_marker.lifetime = ros::Duration();
+        cone_marker.id = cone_id;
+        cone_marker.type = visualization_msgs::Marker::SPHERE;
+        cone_marker.action = visualization_msgs::Marker::ADD;
+        cone_marker.pose.position.x = cone_object.location.x;
+        cone_marker.pose.position.y = cone_object.location.y;
+        cone_marker.pose.position.z = 0;

```

Comment:
I understand that it is _easy_ to add the visualisation topics in here, but I'm not sure this is the right place. If we would have endless time I would split all the visualisation things into their own node. But we are on a time budget so let's keep this for now.

Reply:
so should I keep this here for now? Honestly this visualization code/ topic is really only here for debugging so you can replicate my results and quickly see that the track has the correct shape. I am in favor of removing this code once you approve

Reply:
Yea, let's remove it then :) I think it is amazing to see the cones in rviz but I don't think it should be the responsibility of the ros bridge to provide these topics.